app/controllers/conversations_controller.rb

```diff
@@ -9,5 +9,10 @@ def show
     @conversation = Conversation.find(params[:id])
     @message = Message.new
     authorize @conversation, policy_class: MessagingPolicy
+    @unread_notifications = current_user.conversation_notifications(@conversation)
+    if @unread_notifications.any?
+      @unread_notifications.each { |n| n.mark_as_read! }
+      redirect_to conversation_path(@conversation)
+    end
```

Comment:
```suggestion
    unread_notifications = current_user.unread_conversation_notifications(@conversation)
    unread_notifications.each { |n| n.mark_as_read! }
    redirect_to conversation_path(@conversation)
```

Reply:
Isn't this going to create an infinite loop? Shouldn't the redirect only happen if there are no unread conversation notifications that need to be marked as read?

Reply:
Oh, you're absolutely right. Why, then, are we redirecting at all? We aren't showing any UI for read vs. unread messages so the view should look exactly the same.

Reply:
Yeah, you're right, it totally does not need to be in there. 🤦‍♀️ I mistakenly assumed the redirect was necessary in order to make the red dot disappear.